src/cephadm/cephadm.py

```diff
@@ -3071,10 +3073,14 @@ def get_container_mounts(ctx, fsid, daemon_type, daemon_id,
     if daemon_type == 'osd':
         # selinux-policy in the container may not match the host.
         if HostFacts(ctx).selinux_enabled:
-            selinux_folder = '/var/lib/ceph/%s/selinux' % fsid
-            if not os.path.exists(selinux_folder):
-                os.makedirs(selinux_folder, mode=0o755)
-            mounts[selinux_folder] = '/sys/fs/selinux:ro'
+            cluster_dir = f"/var/lib/ceph/{fsid}"
+            selinux_folder = f"{cluster_dir}/selinux"
+            if os.path.exists(cluster_dir):
```

Comment:
This code was being called by ceph-volume and the direcotry `/var/lib/ceph/<fsid>/selinux` was being created even when the root directory did not exist. The new check prevents this from happening.

Reply:
when/why was this happening? During the rollback on an early failure?

Reply:
> when/why was this happening? During the rollback on an early failure?

This code was executed by ceph-volume once cephadm module is enabled during the bootstrap process. ceph-volume was creating the whole path (including `/var/lib/ceph/<fsid>/` if it doesn't exist) as consequence we ended up with a direcotry on `/var/lib/ceph/<fsid>/` even if the rollback mechansim has already deleted the cluster files. To avoid that, the new code only creates the subdir `selinux` in the case the cluster directory `/var/lib/ceph/<fsid>/` already exists.